modules/oxxionRtdProvider.js

```diff
@@ -0,0 +1,108 @@
+import { submodule } from '../src/hook.js'
+import { logInfo } from '../src/utils.js'
+
+const oxxionRtdSearchFor = [ 'adUnitCode', 'auctionId', 'bidder', 'bidderCode', 'bidId', 'cpm', 'creativeId', 'currency', 'width', 'height', 'mediaType', 'netRevenue', 'originalCpm', 'originalCurrency', 'requestId', 'size', 'source', 'status', 'timeToRespond', 'transactionId', 'ttl', 'sizes', 'mediaTypes', 'src', 'userId', 'labelAny', 'adId' ];
+const LOG_PREFIX = 'oxxionRtdProvider submodule: ';
+
+let allAdUnits = [];
+
+/** @type {RtdSubmodule} */
+export const oxxionSubmodule = {
+  name: 'oxxionRtd',
+  init: init,
+  onAuctionEndEvent: onAuctionEnd,
+  getBidRequestData: getAdUnits,
+};
+
+function init(config, userConsent) {
+  if (!config.params || !config.params.domain || !config.params.contexts || !Array.isArray(config.params.contexts)) return false
+  return true;
+}
+
+function getAdUnits(reqBidsConfigObj, callback, config, userConsent) {
+  const reqAdUnits = reqBidsConfigObj.adUnits;
+  if (Array.isArray(reqAdUnits)) {
+    reqAdUnits.forEach(adunit => {
+      if (adunit.mediaTypes.video !== undefined && adunit.mediaTypes.video.context != undefined && config.params.contexts.includes(adunit.mediaTypes.video.context)) { allAdUnits.push(adunit); }
+    });
+  }
+}
+
+function insertVideoTracking(bidResponse, config, maxCpm) {
+  if (bidResponse.mediaType == 'video') {
+    let trackingUrl = getImpUrl(config, bidResponse, maxCpm);
+    if (!trackingUrl) { return; }
+    // Vast Impression URL
+    bidResponse.vastImpUrl = bidResponse.vastUrl !== undefined
+      ? bidResponse.vastImpUrl !== undefined
+        ? trackingUrl + '&url=' + encodeURI(bidResponse.vastImpUrl)
+        : trackingUrl
+      : bidResponse.vastUrl;
```

Comment:
It should be more robust to check with `!bidResponse.vastUrl`. Actually if for any reason "vastUrl" ou vastImpUrl is `""` or `null` it will pass.